diff --git a/dist/index.mjs b/dist/index.mjs
index 5ba9a077425954c8099a26820a3411b017093b0d..6f19e2251ffb3e69abeeadcb16f18399599300c9 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -450,6 +450,10 @@ function verifyIntersections(items) {
           continue;
         if (isBarHasFooStart && isBarHasFooEnd)
           continue;
+        if (isBarHasFooStart && foo.start.offset === foo.end.offset)
+          continue // leading adjacent empty
+        if (isFooHasBarEnd && bar.start.offset === bar.end.offset)
+          continue // trailing adjacent empty
         throw new ShikiError$1(`Decorations ${JSON.stringify(foo.start)} and ${JSON.stringify(bar.start)} intersect.`);
       }
     }
